opm/simulators/wells/GasLiftStage2.cpp

```diff
@@ -1109,20 +1039,22 @@ updateRates(const std::string &well_name)
         if (this->parent.well_state_.wellIsOwned(well.indexOfWell(), well_name)) {
             const auto &well_ecl = well.wellEcl();
             double factor = well_ecl.getEfficiencyFactor();
-            auto& [delta_oil, delta_gas, delta_alq] = delta;
-                    delta_oil = factor * (gi.new_oil_rate - state.oilRate());
-                    delta_gas = factor * (gi.new_gas_rate - state.gasRate());
-                    delta_alq = factor * (gi.alq - state.alq());
+            auto& [delta_oil, delta_gas, delta_water, delta_alq] = delta;
+            delta_oil = factor * (gi.new_oil_rate - state.oilRate());
```

Comment:
This might not be related to this PR, but I have a question to the above line: This `delta_oil` in the previous line is correct for the first call from removeSurplusALQ_(), see line 631. But if `updateRates()` is called a second time within the loop starting at line 610, then `delta_oil` should be calculated according to the previous oil rate and not `state.oilRate()`? Since `state.oilRate()` is fixed within the loop starting at line 610 (it comes from stage 1 and is not changed within stage 2). So we should rater use something like `delta_oil = factor * (gi.new_oil_rate - this->prev_oil_rate[well_name])` where we need to define a new variable `prev_oil_rate` ?

Reply:
I think I missed that we update the `GasLiftWellState` at line 117. So then this probably works correctly after all.